src/components/implementation/cbuf_mgr/default/cbuf_mgr.c

```diff
@@ -73,23 +84,36 @@ struct cbuf_info {
 struct cbuf_meta_range {
 	struct cbuf_meta *m;
 	vaddr_t dest;
-	u32_t low_id;
+	int low_id;
 	struct cbuf_meta_range *next, *prev;
 };
-#define CBUF_META_RANGE_HIGH(cmr) (cmr->low_id + (PAGE_SIZE/sizeof(struct cbuf_meta)))
+#define CBUF_META_RANGE_HIGH(cmr) (cmr->low_id + (int)(PAGE_SIZE/sizeof(struct cbuf_meta)))
 
 struct cbuf_bin {
-	int size;
+	unsigned long size;
 	struct cbuf_info *c;
 };
 
+struct blocked_thd {
+	unsigned short int thd_id;
+	unsigned long request_size;
+	struct blocked_thd *next, *prev;
+	unsigned long long blk_start;
```

Comment:
I guess this is a 'time' field, i suggest using u64_t and saying, blk_start_time in the time.